app/utils/score_utils.py

```diff
@@ -55,3 +55,71 @@ def calculate_accuracy(
         )
     else:
         raise NotImplementedError(f"Unknown mode: {vanilla_mode}")
+
+
+def calculate_grade(
+    *,
+    vanilla_mode: int,
+    mods: int,
+    acc: float,
+    n300: int,
+    n100: int,
+    n50: int,
+    nmiss: int,
+) -> str:
+    objectCount = n300 + n100 + n50 + nmiss
+    shouldUseSilverGrades = (mods & 1049608) > 0
+    if vanilla_mode == 0:  # osu!
```

Comment:
we should use an enum here ideally (there may be one available in the codebase for gamemodes already?)

as a pattern if u find yourself writing a comment to explain a constant then it's probably a sign it can be abstractly into some more broad "configuration" or "named" state like a const or an enum